components/eam/src/physics/cam/zm_conv.F90

```diff
@@ -886,8 +1544,152 @@ subroutine zm_convr(lchnk   ,ncol    , &
+           microp_st%difm  (ideep(i),k) = loc_microp_st%difm  (i,k)
+           microp_st%cmei  (ideep(i),k) = loc_microp_st%cmei  (i,k)
+
+           microp_st%nuclin(ideep(i),k) = loc_microp_st%nuclin(i,k)
+           microp_st%autoin(ideep(i),k) = loc_microp_st%autoin(i,k)
+           microp_st%accsin(ideep(i),k) = loc_microp_st%accsin(i,k)
+           microp_st%hmpin (ideep(i),k) = loc_microp_st%hmpin (i,k)
+           microp_st%difn  (ideep(i),k) = loc_microp_st%difn  (i,k)
+
+           microp_st%trspcm(ideep(i),k) = loc_microp_st%trspcm(i,k)
+           microp_st%trspcn(ideep(i),k) = loc_microp_st%trspcn(i,k)
+           microp_st%trspim(ideep(i),k) = loc_microp_st%trspim(i,k)
+           microp_st%trspin(ideep(i),k) = loc_microp_st%trspin(i,k)
+
+           microp_st%accgrm(ideep(i),k) = loc_microp_st%accgrm(i,k)
+           microp_st%accglm(ideep(i),k) = loc_microp_st%accglm(i,k)
+           microp_st%accgslm(ideep(i),k)= loc_microp_st%accgslm(i,k)
+           microp_st%accgsrm(ideep(i),k)= loc_microp_st%accgsrm(i,k)
+           microp_st%accgirm(ideep(i),k)= loc_microp_st%accgirm(i,k)
+           microp_st%accgrim(ideep(i),k)= loc_microp_st%accgrim(i,k)
+           microp_st%accgrsm(ideep(i),k)= loc_microp_st%accgrsm(i,k)
+
+           microp_st%accgsln(ideep(i),k)= loc_microp_st%accgsln(i,k)
+           microp_st%accgsrn(ideep(i),k)= loc_microp_st%accgsrn(i,k)
+           microp_st%accgirn(ideep(i),k)= loc_microp_st%accgirn(i,k)
+
+           microp_st%accsrim(ideep(i),k)= loc_microp_st%accsrim(i,k)
+           microp_st%acciglm(ideep(i),k)= loc_microp_st%acciglm(i,k)
+           microp_st%accigrm(ideep(i),k)= loc_microp_st%accigrm(i,k)
+           microp_st%accsirm(ideep(i),k)= loc_microp_st%accsirm(i,k)
+
+           microp_st%accigln(ideep(i),k)= loc_microp_st%accigln(i,k)
+           microp_st%accigrn(ideep(i),k)= loc_microp_st%accigrn(i,k)
+           microp_st%accsirn(ideep(i),k)= loc_microp_st%accsirn(i,k)
+           microp_st%accgln(ideep(i),k) = loc_microp_st%accgln(i,k)
+           microp_st%accgrn(ideep(i),k) = loc_microp_st%accgrn(i,k)
+
+           microp_st%accilm(ideep(i),k) = loc_microp_st%accilm(i,k)
+           microp_st%acciln(ideep(i),k) = loc_microp_st%acciln(i,k)
+
+           microp_st%fallrm(ideep(i),k) = loc_microp_st%fallrm(i,k)
+           microp_st%fallsm(ideep(i),k) = loc_microp_st%fallsm(i,k)
+           microp_st%fallgm(ideep(i),k) = loc_microp_st%fallgm(i,k)
+           microp_st%fallrn(ideep(i),k) = loc_microp_st%fallrn(i,k)
+           microp_st%fallsn(ideep(i),k) = loc_microp_st%fallsn(i,k)
+           microp_st%fallgn(ideep(i),k) = loc_microp_st%fallgn(i,k)
+           microp_st%fhmrm (ideep(i),k) = loc_microp_st%fhmrm (i,k)
+         end if
       end do
    end do
+
+   if (zm_microp) then
+     do k = msg + 1,pver
+       do i = 1,ncol
+
+         !Interpolate variable from interface to mid-layer and convert it from units of "kg/kg" to "g/m3"
```

Comment:
FYI, this is a crude interpolation that won't work well for a highly stretched grid. 
Also, the comment mentions a unit conversion, but the actual code doesn't seem to do any unit conversion until the 3rd block down. I would suggest putting the comment directly adajacent to the relevant block for better readability.

Reply:
The comment is moved to the relevant block. The crude interpolation is only used for output, which is not used in model integration.